File: 프로그래머스/2/84512. 모음 사전/모음 사전.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int cnt = -1;
int answer = 0;
string vowel = "AEIOU";

void DFS(string word, string target){
    cnt++;
    
    if(word == target){
        answer = cnt;
        return;
    }
    
    if(word.length() >= 5)
        return;
    
    for(int i=0; i<vowel.length(); i++)
        DFS(word+vowel[i], target);
}

int solution(string word) {
    DFS("", word);
    
    return answer;
}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.js
function solution(word) {
    var answer = 0;
    let cnt = -1;
    
    let vowel = "AEIOU";
    
    function DFS(tmp){
        cnt++;
        
        if(tmp === word){
            answer = cnt;
            return;
        }
        if(tmp.length >= 5)
            return;
        
        for(let i=0; i<vowel.length; i++)
            DFS(tmp+vowel[i]);
    }
    
    DFS("");
    
    return answer;
}